src/link/MachO/Dylib.zig

```diff
@@ -82,22 +123,22 @@ pub fn parse(self: *Dylib) !void {
         return error.MalformedDylib;
     }
 
-    const this_arch: std.Target.Cpu.Arch = switch (self.header.?.cputype) {
-        macho.CPU_TYPE_ARM64 => .aarch64,
-        macho.CPU_TYPE_X86_64 => .x86_64,
-        else => |value| {
-            log.err("unsupported cpu architecture 0x{x}", .{value});
-            return error.UnsupportedCpuArchitecture;
+    const this_arch: std.Target.Cpu.Arch = decodeArch(self.header.?.cputype) catch |err| switch (err) {
+        error.UnsupportedCpuArchitecture => |e| {
+            log.err("unsupported cpu architecture 0x{x}", .{self.header.?.cputype});
+            return e;
         },
+        else => |e| return e,
     };
+
     if (this_arch != self.arch.?) {
         log.err("mismatched cpu architecture: expected {s}, found {s}", .{ self.arch.?, this_arch });
         return error.MismatchedCpuArchitecture;
     }
 
     try self.readLoadCommands(reader);
     try self.parseId();
-    try self.parseSymbols();
+    try self.parseSymbols(library_offset);
```

Comment:
Given that we may want to parse additional file contents in the future, I think it would make more sense to preserve the offset in `Dylib` and add it whenever we `pread` from the file.